subsys/net/ip/6lo.c

```diff
@@ -1431,3 +1577,24 @@ bool net_6lo_uncompress(struct net_pkt *pkt)
 
 	return true;
 }
+
+int net_6lo_uncompress_hdr_diff(struct net_pkt *pkt)
+{
+	if ((pkt->frags->data[0] & NET_6LO_DISPATCH_IPHC) ==
+	    NET_6LO_DISPATCH_IPHC) {
+		int len;
+
+		if (!uncompress_IPHC_header(pkt, true, &len)) {
+			return INT_MAX;
+		}
+
+		return len;
+	} else if ((pkt->frags->data[0] & NET_6LO_DISPATCH_IPV6) ==
+		   NET_6LO_DISPATCH_IPV6) {
+		return -1;
+	}
+
+	NET_WARN("Nothing to decompress");
```

Comment:
I'm getting  quite a lot of these in my logs. Is a presence of non-compressed header so unusual that it needs a warning level?

Reply:
A forgotten message for debugging, sorry